### PR TITLE
MIR-1412 Show MD5 sum for each file in the derivate file list

### PR DESCRIPTION
**Pull request by levshyn**
The list of derivate files should have the ability to display the MD5 amount

[MIR-1412](https://mycore.atlassian.net/browse/MIR-1412).

[MIR-1412]: https://mycore.atlassian.net/browse/MIR-1412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mir-layout/src/main/resources/META-INF/resources/mir-layout/scss/common/mir_browse.scss [hunk 1]
```diff
@@ -143,4 +143,10 @@ ul ul span.cbNum {
         word-break: break-all;
         padding-right: 20px;
     }
+
+    .md5 {
```
Comment (Possommi) [inline mir-layout/src/main/resources/META-INF/resources/mir-layout/scss/common/mir_browse.scss:147]: I suggest to rename that class to `file_md5` to it follows the same rules as the other file related classed

File: mir-module/src/main/resources/META-INF/resources/hbs/derivate-fileList.hbs [hunk 2]
```diff
@@ -41,6 +41,8 @@
                 <a href="{{href}}">
                   {{name}}
                 </a>
+                <!-- Show md5 sum -->
```
Comment (Possommi) [inline mir-module/src/main/resources/META-INF/resources/hbs/derivate-fileList.hbs:44]: Comment should be removed

File: mir-module/src/main/resources/META-INF/resources/hbs/derivate-fileList.hbs [hunk 3]
```diff
@@ -41,6 +41,8 @@
...
+                <div class="md5 hidden">{{getI18n "mir.derivate.file.MD5"}} {{md5}}</div>
```
Comment (Possommi) [inline mir-module/src/main/resources/META-INF/resources/hbs/derivate-fileList.hbs:45]: If `{{getI18n "mir.derivate.file.MD5"}}` is wrapped in a `<span class="file_md5_label"/>` element other applications are able to hide that part of the content.

**Review by Possommi — CHANGES_REQUESTED** [review]

File: mir-layout/src/main/resources/META-INF/resources/mir-layout/scss/common/mir_browse.scss (hunk 1 above)
Comment (levshyn) [reply mir-layout/src/main/resources/META-INF/resources/mir-layout/scss/common/mir_browse.scss]: fixed

**Review by Possommi — APPROVED** [review]

File: mir-module/src/main/resources/META-INF/resources/js/mir/base.js [hunk 4]
```diff
@@ -329,6 +329,38 @@
...
+        if ($(md5ElementSelector)) {
+            let i18nKey = '';
+            if ($(md5ElementSelector).hasClass('hidden')) {
+                i18nKey = 'component.mods.metaData.options.MD5.show';
+            } else {
+                i18nKey = 'component.mods.metaData.options.MD5.hide';
+            }
+            getI18n(i18nKey, toggleMD5LinkElement);
+        }
+    });
```
Comment (golsch) [inline mir-module/src/main/resources/META-INF/resources/js/mir/base.js:362]: use 2 spaces. Encapsulate the code to improve overview (IIFE).

File: mir-module/src/main/resources/META-INF/resources/js/mir/base.js [hunk 5]
```diff
@@ -329,6 +329,38 @@
...
+    // Getting i18n translation for the link to switch state 'show/hide MD5 amounts' depending on the visibility of the
+    // message with this amount when clicking on the 'derivateActionDropdownToggleButton' element
+    $(derivateActionDropdownToggleButton).click((evt) => {
+        if ($(md5ElementSelector)) {
+            let i18nKey = '';
+            if ($(md5ElementSelector).hasClass('hidden')) {
+                i18nKey = 'component.mods.metaData.options.MD5.show';
+            } else {
+                i18nKey = 'component.mods.metaData.options.MD5.hide';
+            }
```
Comment (golsch) [inline mir-module/src/main/resources/META-INF/resources/js/mir/base.js:359]:
```
const i18nKey = md5Elements.hasClass('hidden') 
  ? 'component.mods.metaData.options.MD5.show' 
  : 'component.mods.metaData.options.MD5.hide';
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/base.js [hunk 6]
```diff
@@ -329,6 +329,38 @@
...
+
+        elements.forEach(element => {
+            element.classList.toggle('hidden');
+        });
+    });
+
+    // Getting i18n translation for the link to switch state 'show/hide MD5 amounts' depending on the visibility of the
+    // message with this amount when clicking on the 'derivateActionDropdownToggleButton' element
+    $(derivateActionDropdownToggleButton).click((evt) => {
+        if ($(md5ElementSelector)) {
```
Comment (golsch) [inline mir-module/src/main/resources/META-INF/resources/js/mir/base.js:353]: use `$(md5ElementSelector).length > 0`, because this check is always true.

**Review by golsch — CHANGES_REQUESTED** [review]
change target branch version to 2024.06.x

**Comment by Possommi** [comment]
> change target branch version to 2024.06.x

but we need this in 2023.06.x

File: mir-module/src/main/resources/META-INF/resources/js/mir/base.js (hunk 4 above)
Comment (levshyn) [reply mir-module/src/main/resources/META-INF/resources/js/mir/base.js]: fixed

File: mir-module/src/main/resources/META-INF/resources/js/mir/base.js (hunk 5 above)
Comment (levshyn) [reply mir-module/src/main/resources/META-INF/resources/js/mir/base.js]: fixed

**Review by kkrebs — APPROVED** [review]

**Pull request merged**